### ISSUE TITLE
Change the use of mutable function values for default arguments

### ISSUE THREAD

**Issue by ysuarez** [mjordan/islandora_workbench#657]
Suggesting a Python best practice, and for reference check:

Beyond the Basic Stuff with Python
by Al Sweigart 
https://inventwithpython.com/beyond/

Ch 8: Common Python Gotchas
https://inventwithpython.com/beyond/chapter8.html

"...
You should never set a mutable object, such as a list or dictionary,
as a default argument. 
...
But using a mutable object, such as a list or dictionary, for the default argument has a subtle problem: the list is created when the function’s def statement executes, not each time the function is called. 
...
If you need to use a list or dictionary as a default argument, the pythonic solution is to set the default argument to None. 
...
Mutable data types include lists, dictionaries, sets, and objects made from the class statement. Don’t put objects of these types as default arguments in a def statement.
..."

Without this change all calls to issue_request() will share the same headers and query dictionaries. Though this could be what you want?

Also, I can move up or down where I declare the new / fresh dictionaries for headers and query variables in issue_request(), and for media_use_tids in get_node_media_ids().

Let me know what you prefer.

**Comment by ysuarez** [comment]
I did some testing of a potential PR, but I got errors. 

My first impression is that the original coding approach had side affects that were expected in the rest of the code. I will have to do more research and testing.

You can see the errors in my draft PR #658, but the last error in the stack trace is

```
AttributeError: 'NoneType' object has no attribute 'update'
```

**Comment by ysuarez** [comment]
I think I found my error. I had to initialize one the `headers` variable earlier in the function.

I am ready for feedback for this issue and PR.

**Comment by mjordan** [comment]
Closed with c989e39fcd2e7b4f666eae889e2bbd654b3d03b8. Thanks for catching these @yamil!
